lib/testing/matrix.js

```diff
@@ -0,0 +1,210 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+* Running tests with multiple configurations using varying parameters
+* @constructor
+*/
+class TestMatrix {
+    /**
+    * Builds a new testing Matrix
+    *
+    * @param {Object.<string, string|string[]>} params - Argument of the matrix
+    *
+    * @returns {TestMatrix} new testing Matrix
+    */
+    constructor(params) {
+        /**
+        * Create deep copy for prevent the user to modify the object outside
```

Comment:
Correction
`Create a deep copy to prevent the user from modifying the object from outside the class`